### PR TITLE
Improve handling of OpenAPI descriptions

### PR DESCRIPTION
**Pull request by robert-cronin**
Details can be found at: https://github.com/kubernetes/website/pull/47089

### CODE REVIEW

File: gen-resourcesdocs/pkg/outputs/kwebsite/section.go [hunk 1]
```diff
@@ -19,7 +21,22 @@ type Section struct {
...
+	// Replace code blocks with properly formatted HTML
+	processed := re.ReplaceAllStringFunc(s, func(match string) string {
+		// Remove the backticks and trim whitespace
+		code := strings.TrimSpace(strings.TrimPrefix(strings.TrimSuffix(match, "```"), "```"))
+		// Escape HTML special characters
+		code = html.EscapeString(code)
+		// Preserve line breaks
+		code = strings.ReplaceAll(code, "\n", "<br>")
+		return "<pre><code>" + code + "</code></pre>"
+	})
```
Comment (tengqm) [inline gen-resourcesdocs/pkg/outputs/kwebsite/section.go:37]: I'm move inclined to treat the description text as markdown.

Comment (robert-cronin) [reply gen-resourcesdocs/pkg/outputs/kwebsite/section.go:37]: Thank you for the markdown suggestion @tengqm - could you point me to any examples of this approach in this repo or recommend any Go libraries for implementation? I noticed that [russross/blackfriday](https://github.com/russross/blackfriday) is installed, that would be my first thought but I am open to suggestions.

Comment (tengqm) [reply gen-resourcesdocs/pkg/outputs/kwebsite/section.go:37]: We use [goldmark](https://github.com/yuin/goldmark) in other generators.

Comment (robert-cronin) [reply gen-resourcesdocs/pkg/outputs/kwebsite/section.go]: Thanks for the guidance @tengqm, I have replaced the html specific rendering logic with a goldmark Parser.

**Comment by tengqm** [comment]
/approve

Please resolve the conflicts ...

**Comment by tengqm** [comment]
> I believe we will still need [kubernetes/kube-openapi#482](https://github.com/kubernetes/kube-openapi/pull/482) in order to have the code render correctly with line breaks.

The line breaks are defects in the upstream source code comments. I don't think we need to
adapt the parser to bad comment formats.

**Comment by robert-cronin** [comment]
> /approve
> 
> Please resolve the conflicts ...

Thank you for the approve @tengqm, I've resolved the conflicts and rebased onto master.

**Pull request merged**